pkg/ecdsa/dkg/member.go

```diff
@@ -111,24 +126,199 @@ func (skgm *symmetricKeyGeneratingMember) MarkInactiveMembers(
 	filter.FlushInactiveMembers()
 }
 
+// initializeTssRoundOne returns a member to perform next protocol operations.
+func (skgm *symmetricKeyGeneratingMember) initializeTssRoundOne() *tssRoundOneMember {
+	tssPartyID, groupTssPartiesIDs := generateTssPartiesIDs(
+		skgm.id,
+		skgm.group.MemberIDs(),
+	)
+
+	tssParameters := tss.NewParameters(
+		tss.EC(),
+		tss.NewPeerContext(tss.SortPartyIDs(groupTssPartiesIDs)),
+		tssPartyID,
+		len(groupTssPartiesIDs),
+		skgm.group.DishonestThreshold(),
```

Comment:
Are we absolutely sure this function accepts the dishonest threshold and not the honest threshold?

Reply:
Without digging too deeply into how we're defining `DishonestThreshold`, if, say, we want to represent a 51-of-100, the `threshold` param to `tss.NewParameters` needs to be `50`. If we want to represent a 3-of-3, it needs to be `2`.

Same thing here: https://github.com/keep-network/keep-ecdsa/blob/5031a0a8c52c2495644a2d94efe9a3efd38f2c80/pkg/ecdsa/tss/signing.go#L126

Reply:
So this should be `HonestThreshold-1` right?

Reply:
Yeah!

I looked up the definition of DishonestThreshold, and something seems up.

```
// HonestThreshold is the minimum number of active participants behaving
// according to the protocol needed to generate a new relay entry.
```

So for a 51-of-100, that reads like it should be 51. Then, DishonestThreshold is:

```
// DishonestThreshold is the maximum number of misbehaving participants for
// which it is still possible to generate a signature.
```
In a 51-of-100, you can have 49 participants misbehave and still generate a sig (so long as the other 51 are all properly behaving), which follow's it's code definition: `cc.GroupSize - cc.HonestThreshold`

In the literature, (and as far as I can tell in tss-lib), this is how the threshold param is defined: https://eprint.iacr.org/2019/114.pdf

> A threshold signature scheme enables n parties to share the power to issue digital signatures
under a single public key. A threshold t is specified such that any subset of t + 1 players can
jointly sign, but any smaller subset cannot.

We want to be super sure. I'll do some tss-lib specific testing but it doesn't have to block this PR. Can always follow up with a PR if `honest-1` was wrong

Reply:
See 6e90f79d